app/services/optout/OptOutService.scala

```diff
@@ -134,25 +160,46 @@ class OptOutService @Inject()(itsaStatusUpdateConnector: ITSAStatusUpdateConnect
     }
   }
 
-  private def makeOptOutUpdateRequestForOneYear(optOutProposition: OptOutProposition)(implicit
-                                                                                      user: MtdItUser[_],
-                                                                                      shc: HeaderCarrier,
-                                                                                      ec: ExecutionContext): Future[OptOutUpdateResponse] = {
-    val intent = optOutProposition.availableOptOutYears.head
-    makeOptOutUpdateRequest(optOutProposition, intent)
-  }
 
   def makeOptOutUpdateRequest(optOutProposition: OptOutProposition, intent: OptOutTaxYear)(implicit user: MtdItUser[_],
-                                                                                           shc: HeaderCarrier, ec: ExecutionContext): Future[OptOutUpdateResponse] = {
+                                                                                           hc: HeaderCarrier, ec: ExecutionContext): Future[OptOutUpdateResponse] = {
 
     val optOutYearsToUpdate = optOutProposition.optOutYearsToUpdate(intent)
 
     val responses: Seq[Future[OptOutUpdateResponse]] = optOutYearsToUpdate.map(optOutYear =>
       itsaStatusUpdateConnector.requestOptOutForTaxYear(optOutYear.taxYear, user.nino, optOutUpdateReason))
     combineByReturningAnyFailureFirstOrAnySuccess(responses)
   }
+
+  def optOutConfirmedPageViewModel()(implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Option[ConfirmedOptOutViewModel]] = {
+    setupOptOutProposition().map {
+      case optOutProposition@OptOutProposition(_, currentTaxYear, nextTaxYear) if optOutProposition.isOneYearOptOut =>
+        val optOutYear = optOutProposition.availableOptOutYears.head
+
+        optOutYear match {
```

Comment:
lines 179 to 188 should move inside OptOutProposition, happy to pair on this

Reply:
done